src/runtime/queue.c

```diff
@@ -0,0 +1,24 @@
+#include <runtime.h>
+
+#define _queue_alloc_size(o) (sizeof(struct queue) + queue_data_size(o))
+
+/* will round up size to next power-of-2 */
+queue allocate_queue(heap h, u64 size)
+{
+    if (size == 0)
+        return INVALID_ADDRESS;
+    int order = find_order(size);
+    queue q = allocate(h, _queue_alloc_size(order));
+    if (q == INVALID_ADDRESS)
+        return q;
+    void *buf = ((void *)q) + sizeof(struct queue);
```

Comment:
in some world, it would be nice to have the body line-aligned

Reply:
made a statically-defined default cacheline size of 64 and aligned to it

Are there x86_64 implementations with other sizes? We could probably find it in a cpuid leaf if we really cared to...

Reply:
looking right now apparently "netburst" machines have a 128byte line, but apparently its somehow split?

since its an optimization, fixed 64 byte alignment is just fine...i'm not even sure if it was
worth bothering you with the idea.

but yeah, I think there are probably other cases where cache aligning some object will be helpful